### PR TITLE
obs 25.0.1

### PR DESCRIPTION
**Pull request by eszlari**

### CODE REVIEW

**Comment by dgcampea** [comment]
That release is a hotfix for Mac systems, brings no value to Linux. And #47 is not fixed with this.

**Comment by eszlari** [comment]
I have reverted the pulseaudio workaround, it's fixed since https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/commit/c364a33322271c3185975369ece65b99ada07c42.

**Comment by eszlari** [comment]
Seems like the runtime has not yet been updated on Flathub?! Luckily, I have add `-DENABLE_PULSEAUDIO=ON` to make it fail explicitly.

**Comment by eszlari** [comment]
Is a rebuild of `org.kde.Sdk//5.14` needed to get the fix?

**Comment by TingPing** [comment]
> Is a rebuild of `org.kde.Sdk//5.14` needed to get the fix?

Yeah.

File: com.obsproject.Studio.json [hunk 1]
```diff
@@ -159,7 +159,9 @@
         "-DUNIX_STRUCTURE=1",
         "-DUSE_XDG=1",
         "-DDISABLE_ALSA=1",
-        "-DDISABLE_JACK=1"
+        "-DDISABLE_JACK=1",
+        "-DENABLE_PULSEAUDIO=ON",
+        "-DOBS_VERSION_OVERRIDE=24.0.6"
```
Comment (dgcampea) [inline com.obsproject.Studio.json]: why override version?

Comment (eszlari) [reply com.obsproject.Studio.json]: To get ride of "-modified" in the version string (e.g. "24.0.5-modified", see https://flathub.org/apps/details/com.obsproject.Studio).

Comment (dgcampea) [reply com.obsproject.Studio.json]: I wouldn't change that. "-modified" might be how OBS team distinguishes official builds.

Comment (eszlari) [reply com.obsproject.Studio.json]: There are no official Linux builds. Distros set that variable:

Debian:
https://salsa.debian.org/multimedia-team/obs-studio/-/blob/master/debian/rules

Arch:
https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/obs-studio

**Comment by eszlari** [comment]
should fix https://github.com/flathub/com.obsproject.Studio/issues/39 as well

File: com.obsproject.Studio.json [hunk 2]
```diff
@@ -147,6 +147,29 @@
             }
         ]
     },
+    {
+      "name": "mbedtls",
```
Comment (TingPing) [inline com.obsproject.Studio.json]: This is a non-optional dep of obs now?

Comment (eszlari) [reply com.obsproject.Studio.json]: Yes.

Comment (dgcampea) [reply com.obsproject.Studio.json]: https://obsproject.com/wiki/install-instructions#linux-build-directions
Required if users want to stream on Facebook

Comment (eszlari) [reply com.obsproject.Studio.json]: Even with `-DWITH_RTMPS=OFF` it fails to compile:
```

[194/375] Building C object plugins/obs-outputs/CMakeFiles/obs-outputs.dir/librtmp/rtmp.c.o
FAILED: plugins/obs-outputs/CMakeFiles/obs-outputs.dir/librtmp/rtmp.c.o 
/run/ccache/bin/cc -DFTL_STATIC_COMPILE -DHAVE_OBSCONFIG_H -DNO_CRYPTO -DUSE_XDG -Dobs_outputs_EXPORTS -Iplugins/obs-outputs/ftl-sdk/libftl/posix -Iplugins/obs-outputs/ftl-sdk/libftl -Iplugins/obs-outputs/config -Ilibobs -Iconfig -Ideps/jansson/src -Ideps/jansson/include -Wall -Wextra -Wvla -Wno-unused-function -Werror-implicit-function-declaration -Wno-missing-braces -Wno-missing-field-initializers -O2 -g -pipe -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -std=gnu99 -fno-strict-aliasing -O3 -DNDEBUG -fPIC   -mmmx -msse -msse2 -pthread -fvisibility=hidden -MD -MT plugins/obs-outputs/CMakeFiles/obs-outputs.dir/librtmp/rtmp.c.o -MF plugins/obs-outputs/CMakeFiles/obs-outputs.dir/librtmp/rtmp.c.o.d -o plugins/obs-outputs/CMakeFiles/obs-outputs.dir/librtmp/rtmp.c.o   -c plugins/obs-outputs/librtmp/rtmp.c
plugins/obs-outputs/librtmp/rtmp.c: In function ‘RTMP_TLS_LoadCerts’:
plugins/obs-outputs/librtmp/rtmp.c:287:26: warning: unused parameter ‘r’ [-Wunused-parameter]
  287 | RTMP_TLS_LoadCerts(RTMP *r) {
      |                    ~~~~~~^
plugins/obs-outputs/librtmp/rtmp.c: In function ‘RTMP_TLS_Init’:
plugins/obs-outputs/librtmp/rtmp.c:367:21: warning: unused parameter ‘r’ [-Wunused-parameter]
  367 | RTMP_TLS_Init(RTMP *r)
      |               ~~~~~~^
plugins/obs-outputs/librtmp/rtmp.c: In function ‘RTMP_TLS_Free’:
plugins/obs-outputs/librtmp/rtmp.c:414:21: warning: unused parameter ‘r’ [-Wunused-parameter]
  414 | RTMP_TLS_Free(RTMP *r) {
      |               ~~~~~~^
plugins/obs-outputs/librtmp/rtmp.c: In function ‘b64enc’:
plugins/obs-outputs/librtmp/rtmp.c:2547:9: error: unknown type name ‘base64_encodestate’
 2547 |         base64_encodestate state;
      |         ^~~~~~~~~~~~~~~~~~
plugins/obs-outputs/librtmp/rtmp.c:2549:9: error: implicit declaration of function ‘base64_init_encodestate’ [-Werror=implicit-function-declaration]
 2549 |         base64_init_encodestate(&state);
      |         ^~~~~~~~~~~~~~~~~~~~~~~
plugins/obs-outputs/librtmp/rtmp.c:2550:19: error: implicit declaration of function ‘base64_encode_block’ [-Werror=implicit-function-declaration]
 2550 |         output += base64_encode_block((const char *)input, length, output, &state);
      |                   ^~~~~~~~~~~~~~~~~~~
plugins/obs-outputs/librtmp/rtmp.c:2551:9: error: implicit declaration of function ‘base64_encode_blockend’ [-Werror=implicit-function-declaration]
 2551 |         base64_encode_blockend(output, &state);
      |         ^~~~~~~~~~~~~~~~~~~~~~
plugins/obs-outputs/librtmp/rtmp.c: In function ‘PublisherAuth’:
plugins/obs-outputs/librtmp/rtmp.c:2639:30: error: ‘MD5_DIGEST_LENGTH’ undeclared (first use in this function)
 2639 |     unsigned char md5sum_val[MD5_DIGEST_LENGTH+1];
      |                              ^~~~~~~~~~~~~~~~~
plugins/obs-outputs/librtmp/rtmp.c:2639:30: note: each undeclared identifier is reported only once for each function it appears in
plugins/obs-outputs/librtmp/rtmp.c:2640:5: error: unknown type name ‘MD5_CTX’
 2640 |     MD5_CTX md5ctx;
      |     ^~~~~~~
plugins/obs-outputs/librtmp/rtmp.c:2731:13: error: implicit declaration of function ‘MD5_Init’ [-Werror=implicit-function-declaration]
 2731 |             MD5_Init(&md5ctx);
      |             ^~~~~~~~
plugins/obs-outputs/librtmp/rtmp.c:2732:13: error: implicit declaration of function ‘MD5_Update’ [-Werror=implicit-function-declaration]
 2732 |             MD5_Update(&md5ctx, user.av_val, user.av_len);
      |             ^~~~~~~~~~
plugins/obs-outputs/librtmp/rtmp.c:2735:13: error: implicit declaration of function ‘MD5_Final’ [-Werror=implicit-function-declaration]
 2735 |             MD5_Final(md5sum_val, &md5ctx);
      |             ^~~~~~~~~
plugins/obs-outputs/librtmp/rtmp.c:2852:62: warning: unused variable ‘hash3’ [-Wunused-variable]
 2852 |             char hash1[HEXHASH_LEN+1], hash2[HEXHASH_LEN+1], hash3[HEXHASH_LEN+1];
      |                                                              ^~~~~
plugins/obs-outputs/librtmp/rtmp.c:2852:40: warning: unused variable ‘hash2’ [-Wunused-variable]
 2852 |             char hash1[HEXHASH_LEN+1], hash2[HEXHASH_LEN+1], hash3[HEXHASH_LEN+1];
      |                                        ^~~~~
plugins/obs-outputs/librtmp/rtmp.c:2852:18: warning: unused variable ‘hash1’ [-Wunused-variable]
 2852 |             char hash1[HEXHASH_LEN+1], hash2[HEXHASH_LEN+1], hash3[HEXHASH_LEN+1];
      |                  ^~~~~
plugins/obs-outputs/librtmp/rtmp.c:2639:19: warning: unused variable ‘md5sum_val’ [-Wunused-variable]
 2639 |     unsigned char md5sum_val[MD5_DIGEST_LENGTH+1];
      |                   ^~~~~~~~~~
cc1: some warnings being treated as errors
[200/375] Building C object plugins/obs-outputs/CMakeFiles/obs-outputs.dir/rtmp-stream.c.o
ninja: build stopped: subcommand failed.
Error: module obs: Child process exited with code 1
```

Comment (eszlari) [reply com.obsproject.Studio.json]: But why not include it? This also fixes https://github.com/flathub/com.obsproject.Studio/issues/39.

Comment (TingPing) [reply com.obsproject.Studio.json]: I'll accept it, but its a bad decision on OBS' part to mix multiple TLS implementations, and it sucks for us to bundle one.

**Pull request merged**